### ISSUE TITLE
setting parameters for default mountainsort3 pipeline

### ISSUE THREAD

**Issue by kc13** [flatironinstitute/mountainsort#38]
Hello MountainSort developers, 

It's been great to get an opportunity to try out this software.   I would be interested in exploring some changes in parameter settings.  For some parameters (e.g., adjacency_radius, samplerate, detect_sign), I have been able to make the adjustments via command line and/or the parameters JSON file.   For other parameters (consolidate_clusters, merge_across_channels, fit_stage, detect_interval, clip_size), I have not had success with command line flags, the params file, or direct adjustments to the pipeline spec through mlpipeline (e.g., for consolidate clusters, I tried adding "param('consolidate_clusters','false');".  

I assume that I am unsuccessful because the terminal output line that starts with "RUNNING mountainsortalg.ms3:" prints the command line with flags that do not reflect my requested settings (e.g., I'll see "consolidate_clusters=true" after my attempts to set it to false). 

To summarize as a question: Is it possible for a user to set these parameters?  If so, how?   

Apologies in advance if I'm missing something obvious.  Also, if there is still a need for users to volunteer matlab/python/other scripts for tasks related to MountainSort use, I may be able to contribute to that, given some info on what specifically is needed (and what language is preferred). 

Thanks very much -- 
-Karin

**Comment by llobetv** [comment]
Hi, 

if you still need it, there is 2 things to change in  mountainsort3.mlp : 
in this list of params add the params that you want and its default value (i add clip_size, detect_interval and consolidation_factor)

\nparam('samplerate',30000);\nparam('freq_min',300);\nparam('freq_max',6000);\nparam('freq_wid',1000);\nparam('whiten','true');\nparam('detect_threshold',3);\nparam('detect_sign',0);\nparam('adjacency_radius',-1);\nparam('curate','false');\nparam('consolidation_factor',0.9);\nparam('detect_interval', 5);\nparam('clip_size',50);

and then change "var p"  which contain var that are send to sort process. ( i also add clip_size, detect_interval and consolidation_factor). 

var p={\n    detect_threshold:pp.detect_threshold,\n    detect_sign:pp.detect_sign,\n    adjacency_radius:pp.adjacency_radius\n,\n    clip_size:pp.clip_size\n,\n    detect_interval:pp.detect_interval\n,\n    consolidation_factor:pp.consolidation_factor\n };

And then you can change them with .json file.

PS : for MountainSort developers, you're soft is really nice, thank you for that but it's hard to manipulate for newcomers and doc is a bit poor.


Best,

Victor

**Comment by magland** [comment]
Thanks very much Victor for the feedback. I will note that we have a new release of the software (pre-release already available) that should be more intuitive.
https://github.com/flatironinstitute/mountainlab-js
